manim/mobject/svg/svg_mobject.py

```diff
@@ -559,93 +666,155 @@ def generate_points(self):
+                self.add_line_to(p)
+            return
 
+        elif command in ["H", "V", "L"]:  # lineto of any kind
             for p in new_points:
-                if isLower:
-                    # Treat everything as relative line-to until empty
-                    p[0] += self.points[-1, 0]
-                    p[1] += self.points[-1, 1]
                 self.add_line_to(p)
             return
 
-        elif command in ["L", "H", "V"]:  # lineto
-            if command == "H":
-                new_points[0, 1] = points[-1, 1]
-            elif command == "V":
-                if isLower:
-                    new_points[0, 0] -= points[-1, 0]
-                    new_points[0, 0] += points[-1, 1]
-                new_points[0, 1] = new_points[0, 0]
-                new_points[0, 0] = points[-1, 0]
-            self.add_line_to(new_points[0])
+        elif command == "C":  # Cubic
+            # points must be added in groups of 3.
+            for i in range(0, len(new_points), 3):
+                self.add_cubic_bezier_curve_to(*new_points[i : i + 3])
             return
 
-        if command == "C":  # curveto
-            pass  # Yay! No action required
-        elif command in ["S", "T"]:  # smooth curveto
-            self.add_smooth_curve_to(*new_points)
-            # handle1 = points[-1] + (points[-1] - points[-2])
-            # new_points = np.append([handle1], new_points, axis=0)
+        elif command == "S":  # Smooth cubic
+            prev_handle = start_point
+            if prev_command.upper() in ["C", "S"]:
+                prev_handle = self.points[-2]
+            for i in range(0, len(new_points), 2):
+                new_handle = 2 * start_point - prev_handle
+                self.add_cubic_bezier_curve_to(
+                    new_handle, new_points[i], new_points[i + 1]
+                )
+                start_point = new_points[i + 1]
+                prev_handle = new_points[i]
             return
+
         elif command == "Q":  # quadratic Bezier curve
-            # TODO, this is a suboptimal approximation
-            new_points = np.append([new_points[0]], new_points, axis=0)
+            for i in range(0, len(new_points), 2):
+                self.add_quadratic_bezier_curve_to(new_points[i], new_points[i + 1])
+            return
+
+        elif command == "T":  # smooth quadratic
+            prev_quad_handle = start_point
+            if prev_command.upper() in ["Q", "T"]:
+                # because of the conversion from quadratic to cubic,
+                # our actual previous handle was 3/2 in the direction of p[-2] from p[-1]
+                prev_quad_handle = 1.5 * self.points[-2] - 0.5 * self.points[-1]
+            for p in new_points:
+                new_quad_handle = 2 * start_point - prev_quad_handle
+                self.add_quadratic_bezier_curve_to(new_quad_handle, p)
+                start_point = p
+                prev_quad_handle = new_quad_handle
```

Comment:
Nice improvement!

Reply:
Isn't it great that N-degree Beziers are a subset of N+1 degree Beziers?